File: src/main/java/common/entities/Entity.java
package common.entities;

import common.world.Direction;
import talaria.common.entity.AdaptiveNetworkEntity;


public abstract class Entity extends AdaptiveNetworkEntity implements IEntity {
    private Direction facing;

    public Entity(String name, int health, int maxHealth) {
        this.putAttribute("name", name);
        this.putAttribute("health", health);
        this.putAttribute("maxHealth", maxHealth);
        this.facing = Direction.LEFT;
    }

    public String getName() {
        return this.<String>getAttribute("name").get();
    }

    public void setName(String name) {
        this.putAttribute("name", name);
    }

    public int getHealth() {
        return this.<Integer>getAttribute("health").get();
    }

    public void setHealth(int health) {
        this.putAttribute("health", health);
    }

    public int getMaxHealth() {
        return this.<Integer>getAttribute("maxHealth").get();
    }

    public void setMaxHealth(int maxHealth) {
        this.putAttribute("maxHealth", maxHealth);
    }

    @Override
    public void onCollision() {
    }

    public Direction getFacing() {
        return facing;
    }

    public void setFacing(Direction facing) {
        this.facing = facing;
    }

    public abstract void update();
}

File: src/main/kotlin/common/entities/Entity.kt
package common.entities

import common.world.Direction
import bvanseg.talaria.common.entity.AdaptiveNetworkEntity

abstract class Entity(name: String, health: Int, maxHealth: Int) : AdaptiveNetworkEntity() {
    var facing: Direction = Direction.LEFT

    var name: String
        get() = this.getAttribute<String>("name")!!.get()
        set(name) = this.putAttribute("name", name)

    var health: Int
        get() = this.getAttribute<Int>("health")!!.get()
        set(health) = this.putAttribute("health", health)

    var maxHealth: Int
        get() = this.getAttribute<Int>("maxHealth")!!.get()
        set(maxHealth) = this.putAttribute("maxHealth", maxHealth)

    init {
        this.putAttribute("name", name)
        this.putAttribute("health", health)
        this.putAttribute("maxHealth", maxHealth)
        this.facing = Direction.LEFT
    }

    abstract fun onCollision()

    abstract fun update()
}
